src/parquet/column/statistics-test.cc

```diff
@@ -0,0 +1,212 @@
+using schema::PrimitiveNode;
+using schema::GroupNode;
+
+namespace test {
+
+template <typename TestType>
+class TestRowGroupStatistics : public ::testing::Test {
+ public:
+  typedef typename TestType::c_type T;
+
+  void SetUp() {
+    node_ = PrimitiveNode::Make("column", Repetition::REQUIRED, TestType::type_num,
+        LogicalType::NONE, FLBA_LENGTH);
+    schema_ = std::make_shared<ColumnDescriptor>(node_, 0, 0);
+  }
+
+  void GenerateData(int64_t num_values);
+
+  void TestMinMaxEncode() {
+    GenerateData(1000);
+
+    TypedRowGroupStatistics<TestType> statistics1(schema_.get());
+    statistics1.Update(values_ptr_, values_.size(), 0);
+    std::string encoded_min = statistics1.EncodedMin();
+    std::string encoded_max = statistics1.EncodedMax();
+
+    TypedRowGroupStatistics<TestType> statistics2(
+        schema_.get(), encoded_min, encoded_max, values_.size(), 0, 0);
+
+    ASSERT_EQ(encoded_min, statistics2.EncodedMin());
+    ASSERT_EQ(encoded_max, statistics2.EncodedMax());
+    ASSERT_EQ(statistics1.min(), statistics2.min());
+    ASSERT_EQ(statistics1.max(), statistics2.max());
+  }
+
+  void TestReset() {
+    GenerateData(1000);
+
+    TypedRowGroupStatistics<TestType> statistics(schema_.get());
+    statistics.Update(values_ptr_, values_.size(), 0);
+    ASSERT_EQ(values_.size(), statistics.num_values());
+
+    statistics.Reset();
+    ASSERT_EQ(0, statistics.null_count());
+    ASSERT_EQ(0, statistics.num_values());
+    ASSERT_EQ("", statistics.EncodedMin());
+    ASSERT_EQ("", statistics.EncodedMax());
+  }
+
+  void TestMerge() {
+    int num_null[2];
+    random_numbers(2, 42, 0, 100, num_null);
+
+    TypedRowGroupStatistics<TestType> statistics1(schema_.get());
+    GenerateData(1000);
+    statistics1.Update(values_ptr_, values_.size() - num_null[0], num_null[0]);
+
+    TypedRowGroupStatistics<TestType> statistics2(schema_.get());
+    GenerateData(1000);
+    statistics2.Update(values_ptr_, values_.size() - num_null[1], num_null[1]);
+
+    TypedRowGroupStatistics<TestType> total(schema_.get());
+    total.Merge(statistics1);
+    total.Merge(statistics2);
+
+    ASSERT_EQ(num_null[0] + num_null[1], total.null_count());
+    ASSERT_EQ(values_.size() * 2 - num_null[0] - num_null[1], total.num_values());
+    ASSERT_EQ(total.min(), std::min(statistics1.min(), statistics2.min()));
+    ASSERT_EQ(total.max(), std::max(statistics1.max(), statistics2.max()));
+  }
+
+  using TypedStats = TypedRowGroupStatistics<TestType>;
+
+  void TestFullRoundtrip(int64_t num_values, int64_t null_count) {
+    auto node = PrimitiveNode::Make("column", Repetition::OPTIONAL, TestType::type_num,
+        LogicalType::NONE, FLBA_LENGTH);
+    auto gnode = std::static_pointer_cast<GroupNode>(
+        GroupNode::Make("schema", Repetition::REQUIRED, std::vector<NodePtr>({node})));
+
+    GenerateData(num_values);
+
+    // compute statistics for the whole batch
+    ColumnDescriptor descr(node, 1 /* max definition level */, 0);
```

Comment:
This is a little bit unusual; it would be better to do `const int max_def_level = 1;` if you're concerned about readability